subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_internal.h

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SW_SWITCH_RX 0
+#define SW_SWITCH_TX 1
+#define SW_SWITCH_PHY_1M 0
+#define SW_SWITCH_FLAGS_DONTCARE 0
+
+void sw_switch(uint8_t dir_curr, uint8_t dir_next, uint8_t phy_curr, uint8_t flags_curr,
+	       uint8_t phy_next, uint8_t flags_next);
```

Comment:
The order of the arguments is a bit odd as they go:
dir, dir, phy, flags, phy, flags

It would make more sense if it was
dir, phy, flags, dir, phy, flags
or
dir, dir, phy, phy, flags, flags

Or possible using a struct
```
struct sw_switch_param {
    uint8_t dir;
    uint8_t phy;
    uint8_t flags;
};

void sw_switch(struct sw_switch_param curr, struct sw_switch_param next);
```

Reply:
1. See this comment: https://github.com/zephyrproject-rtos/zephyr/pull/36585/commits/6c7bba67f157170e3642cfe9644af09b5c477164#r677372071
2. I've observed that is it kind of a pattern in controller to go for long parameters list instead of structures. @cvinayak can you give more feedback about your choices in this regard?

Reply:
I don't see any comment in your reference :s 

And I agree that it has been a typical pattern in the controller code.

Reply:
Sorry, I've taken link to comment from commit instead of conversation page.
This is correct one: https://github.com/zephyrproject-rtos/zephyr/pull/36585#discussion_r677372071